install-server.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+set -e
+
+echo -e "\n\nEverything is now getting setup. This process will take a few minutes...\n\n"
+
+sudo apt update && sudo apt install gpg jq -y && sudo apt clean
+
+#remove unattended upgrades
+sudo apt-get remove --purge unattended-upgrades || true
+
+#Install tailscale
+curl -fsSL https://tailscale.com/install.sh | sh
+
+#Install nix package manager
+sh <(curl -L https://nixos.org/nix/install) --daemon --yes
+
+# Load the Nix environment
+. /etc/profile.d/nix.sh
+
+# Install packages from a pinned version of Nixpkgs
+nix-env -iA \
+  -f https://github.com/NixOS/nixpkgs/archive/53f3c92b4d1f5f297258954c5026e39c45afd180.tar.gz \
+  libvirt qemu bridge-utils virt-manager dnsmasq
+
+# Find the Nix-installed dnsmasq binary
+DNSMASQ_BIN=$(which dnsmasq)
+LIBVIRTD_BIN=$(which libvirtd)
+
+
+if [[ -z "$DNSMASQ_BIN" || -z "$LIBVIRTD_BIN" ]]; then
+    echo "Error: One or both binaries (dnsmasq, libvirtd) not found after installation!"
+    exit 1
+fi
+
+# Create a systemd service for dnsmasq
+echo "Creating systemd service for dnsmasq..."
+sudo bash -c "cat > /etc/systemd/system/dnsmasq-nix.service" <<EOF
+[Unit]
+Description=DNS caching server (via Nix)
+After=network.target
+
+[Service]
+ExecStart=$DNSMASQ_BIN --no-daemon
```

Comment:
@NichArchA82 I assume DNS Masq is used by the libvirt managed VM's to forward DNS requests? If so, could we configure DNSMasq to use cloudflare/google for DNS resolution? I have had issues particular with Hetzner when hitting an NXDOMAIN error. Cloudflare/Google DNS servers are generally more forgiving and are easily purgeable when issues happen.

`--no-resolv --server=1.1.1.1 --server=8.8.8.8`